service/src/main/java/bio/terra/cbas/runsets/monitoring/SmartRunsPoller.java

```diff
@@ -89,34 +99,32 @@ public List<Run> updateRuns(List<Run> runs) throws Exception {
 
     for (Map.Entry<CbasRunStatus, List<Run>> engineStateEntry : engineStatuses.entrySet()) {
       for (Run r : engineStateEntry.getValue()) {
-        var currentState = engineStateEntry.getKey();
-        if (r.status() != currentState) {
-          Run rWithCurrentStatus = r.withStatus(currentState);
-          if (rWithCurrentStatus.status() == CbasRunStatus.COMPLETE) {
-            // Turn string to an object
-            List<WorkflowOutputDefinition> outputDefinitionList =
-                objectMapper.readValue(
-                    rWithCurrentStatus.runSet().method().outputDefinition(),
-                    new TypeReference<>() {});
-            Object outputs = cromwellService.getOutputs(rWithCurrentStatus.engineId());
-            RecordAttributes outputParamDef =
-                OutputGenerator.buildOutputs(outputDefinitionList, outputs);
-            RecordRequest request = new RecordRequest().attributes(outputParamDef);
-            wdsService.updateRecord(request, r.runSet().method().recordType(), r.recordId());
+        var updatedRunState = engineStateEntry.getKey();
+        if (r.status() != updatedRunState) {
+          if (updatedRunState == CbasRunStatus.COMPLETE) {
+            try {
+              updateOutputAttributes(r);
+            } catch (Exception e) {
+              // log error and mark Run as Failed
+              // TODO: When epic WM-1433 is being worked on, add error message in database stating
+              //  updating output attributes failed for this particular Run.
+              logger.error("Error while updating output attributed for run {}.", r.id(), e);
```

Comment:
Small typo. Also can we include the record we're failing to update? (Warning: I'd double check this suggestion in case I got the name wrong!)
```suggestion
              logger.error("Error while updating attributes for record {} from run {}.", r.recordId() r.id(), e);
```